node_modules/oae-authentication/lib/strategies/shibboleth/init.js

```diff
@@ -84,6 +84,11 @@ module.exports = function(config) {
                 'authoritative': true
             };
 
+            // Set users whose name resembles a Shibboleth identifier as private (eg. starts with http://)
+            if (displayName.indexOf('http://') === 0 || displayName.indexOf('shibboleth!') != -1) {
+                opts.visibility ='private';
+            } 
+
```

Comment:
Cool - I'll have a look at how to do this and will then create a new pull request over there. Meanwhile the issues you've raised here should now all be fixed (I've extended the check to look for emails and added test cases).